### PR TITLE
Refactor the driver

### PR DESCRIPTION
**Pull request by floitsch**
Prepare for bus operations where more than one DS18B20 is on the same 1-wire bus.

### CODE REVIEW

File: examples/shared_bus.toit [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2022 Toitware ApS.
```
Comment (erikcorry) [inline examples/shared_bus.toit:1]:
```suggestion
// Copyright (C) 2023 Toitware ApS.
```

File: src/ds18b20.toit [hunk 2]
```diff
@@ -2,5 +2,147 @@
...
+    owns_bus_ = false
+    is_single_ = false
+    this.id = id
+
+  /**
+  Constructs an instance of the DS18B20 sensor driver for broadcasting.
+  All commands are sent to all devices on the bus.
+
+  If $is_single is true, then the driver assumes there is only one device on the
+    bus. In that case, this instance behaves similar to one constructed with
```
Comment (erikcorry) [inline src/ds18b20.toit:68]:
```suggestion
    bus. In that case, this instance behaves similarly to one constructed with
```

**Comment by erikcorry** [comment]
Also a 2023 copyright on the src/ file.

**Comment by floitsch** [comment]
> 

done.

**Pull request merged**